base/abstractarray.jl

```diff
@@ -373,9 +373,16 @@ zero{T}(x::AbstractArray{T}) = fill!(similar(x), zero(T))
 # While the definitions for LinearFast are all simple enough to inline on their
 # own, LinearSlow's CartesianRange is more complicated and requires explicit
 # inlining.
-start(A::AbstractArray) = (@_inline_meta(); itr = eachindex(A); (itr, start(itr)))
-next(A::AbstractArray,i) = (@_inline_meta(); (idx, s) = next(i[1], i[2]); (A[idx], (i[1], s)))
-done(A::AbstractArray,i) = done(i[1], i[2])
+start(A::AbstractArray) = (@_inline_meta(); _start(linearindexing(A), A))
+_start(::LinearFast, A::AbstractArray) = 1
+_start(::LinearSlow, A::AbstractArray) = (@_inline_meta(); itr = eachindex(A); (itr, start(itr)))
```

Comment:
Right, so this failure was because `convert(UTF16String, UInt16[65 66; 67 68])` calls reshape, so then the method was generalized to accept any AbstractVector.  It'll still fail with non-linear SubArrays (but, really, vector SubArrays should probably be linear, too).

`endof` is documented to return an integer.